src/libraries/Microsoft.PowerFx.Interpreter/RecalcEngine.cs

```diff
@@ -140,19 +140,19 @@ public void UpdateVariable(string name, FormulaValue value)
         /// <param name="parameters">parameters for formula. The fields in the parameter record can 
         /// be acecssed as top-level identifiers in the formula.</param>
         /// <returns>The formula's result.</returns>
-        public FormulaValue Eval(string expressionText, RecordValue parameters = null)
+        public FormulaValue Eval(string expressionText, RecordValue parameters = null, ParserOptions options = null)
         {
-            return EvalAsync(expressionText, CancellationToken.None, parameters).Result;          
+            return EvalAsync(expressionText, CancellationToken.None, parameters, options).Result;          
         }
 
-        public async Task<FormulaValue> EvalAsync(string expressionText, CancellationToken cancel, RecordValue parameters = null)
+        public async Task<FormulaValue> EvalAsync(string expressionText, CancellationToken cancel, RecordValue parameters = null, ParserOptions options = null)
         {
             if (parameters == null)
             {
                 parameters = RecordValue.Empty();
             }
 
-            var check = Check(expressionText, (RecordType)parameters.IRContext.ResultType);
+            var check = Check(expressionText, (RecordType)parameters.IRContext.ResultType, options);
```

Comment:
>Check [](http://example.com/codeflow?start=24&length=5)

I think there's still a followup issue about enforcing / rejecting behavior functions. But we don't have behavior funcs in fx yet (and custom Funcs don't allow specifying behavior properties yet).